test/system/220-healthcheck.bats

```diff
@@ -273,4 +273,182 @@ Log[-1].Output   | \"Uh-oh on stdout!\\\nUh-oh on stderr!\\\n\"
     done
 }
 
+function _create_container_with_health_log_settings {
+    local ctrname="$1"
+    local msg="$2"
+    local format="$3"
+    local flag="$4"
+    local expect="$5"
+    local expect_msg="$6"
+
+    run_podman run -d --name $ctrname   \
+               --health-cmd "echo $msg" \
+               $flag                    \
+               $IMAGE /home/podman/pause
+    cid="$output"
+
+    run_podman inspect $ctrname --format $format
+    is "$output" "$expect" "$expect_msg"
+
+    output=$cid
+}
+
+function _check_health_log {
+    local ctrname="$1"
+    local expect_msg="$2"
+    local expect_count="$3"
+
+    run_podman inspect $ctrname --format "{{.State.Health.Log}}"
+    count=$(grep -co "$expect_msg" <<< "$output")
+    assert "$count" -eq $expect_count "Number of matching health log messages"
+}
+
+@test "podman healthcheck --health-max-log-count default value (5)" {
+    local repeat_count=10
+    local msg="healthmsg-$(random_string)"
+    local ctrname="c-h-$(safename)"
+    _create_container_with_health_log_settings $ctrname $msg "{{.Config.HealthMaxLogCount}}" "" "5" "HealthMaxLogCount is the expected default"
+
+    for i in $(seq 1 $repeat_count);
+    do
+        run_podman healthcheck run $ctrname
+        is "$output" "" "unexpected output from podman healthcheck run (pass $i)"
+    done
+
+    _check_health_log $ctrname $msg 5
+
+    run_podman rm -t 0 -f $ctrname
+}
+
+@test "podman healthcheck --health-max-log-count infinite value (0)" {
+    local repeat_count=10
+    local msg="healthmsg-$(random_string)"
+    local ctrname="c-h-$(safename)"
+    _create_container_with_health_log_settings $ctrname $msg "{{.Config.HealthMaxLogCount}}" "--health-max-log-count 0" "0" "HealthMaxLogCount"
+
+    # This is run 11 times to check that the cap is working.
+    for i in $(seq 0 $repeat_count);
+    do
+        run_podman healthcheck run $ctrname
+        is "$output" "" "unexpected output from podman healthcheck run (pass $i)"
+    done
+
+    # The healthcheck is triggered by the podman when the container is started, but its execution depends on systemd.
+    # And since `run_podman healthcheck run` is also run manually, it will result in two runs.
+    _check_health_log $ctrname $msg 12
+
+    run_podman rm -t 0 -f $ctrname
+}
+
+
+@test "podman healthcheck --health-max-log-count 10" {
+    local repeat_count=10
+    local msg="healthmsg-$(random_string)"
+    local ctrname="c-h-$(safename)"
+    _create_container_with_health_log_settings $ctrname $msg "{{.Config.HealthMaxLogCount}}" "--health-max-log-count  $repeat_count" "$repeat_count" "HealthMaxLogCount"
+
+    # This is run 11 times to check that the cap is working.
```

Comment:
Suggestion: `is run one more time than max-count`. And instead of `0 $repeat_count`, which is too subtle for a quick glance, `$(seq 1 $repeat_count) plusone`